solr/core/src/java/org/apache/solr/handler/component/HttpShardHandler.java

```diff
@@ -148,6 +166,11 @@ public void cancelAll() {
       future.cancel(false);
     }
   }
+  public void cancelAll(boolean mayInterruptIfRunning) {
+    for (Future<ShardResponse> future : pending) {
```

Comment:
Will this interrupt the thread which is waiting for remote shard response?

Reply:
yes